src/plugins/expressions/common/expression_functions/types.ts

```diff
@@ -46,6 +48,11 @@ export interface ExpressionFunctionDefinition<
    */
   name: Name;
 
+  /**
+   * if set to true function will be disabled (but its migrate function will still be available)
+   */
+  disabled?: boolean;
```

Comment:
Do you see any potential overlap with `disabled` and the `hidden`/`internal` functionality we have talked about? 

One idea that was floated was having a list of tags which could be added to any function, which I could see serving a similar purpose to this.

Although I suppose in the case of `disabled` this isn't an opt-in functionality and would always override other tags anyway, so maybe keeping it at the top level is appropriate.

One naming nit: Maybe it's just me, but IMO `enabled: false` reads more clearly than `disabled: true`.

Reply:
yeah i think this should be a separate flag, i also agree its not ideal we need to do this but unless we want to introduce a separate registry for migration function i think we need to do this